pkg/apis/walruscore/v1/template.go

```diff
@@ -41,3 +84,52 @@ type TemplateList struct {
 }
 
 var _ runtime.Object = (*TemplateList)(nil)
+
+// VCSRepository specifies the vcs repository of the template.
+type VCSRepository struct {
+	TokenAuth `json:",inline"`
+
+	// Platform of the vcs repository.
+	//
+	// +k8s:validation:enum=["Github","Gitlab","Gitee"]
+	// +k8s:validation:cel[0]:rule="oldSelf == self"
+	// +k8s:validation:cel[0]:message="immutable field"
+	Platform VCSPlatform `json:"platform"`
+
+	// URL of download the template from vsc repository, may include ref and subpath, e.g. https://github.com/walrus-catalog/terraform-static-redis.
+	//
+	// +k8s:validation:cel[0]:rule="oldSelf == self"
+	// +k8s:validation:cel[0]:message="immutable field"
+	URL string `json:"url"`
+}
+
+// TemplateVersion defines the version of Template.
+type TemplateVersion struct {
+	// Version of the template.
+	//
+	// +optional
+	Version string `json:"version,omitempty"`
+
+	// URL of downloading the template version with ref and subpath.
+	URL string `json:"url"`
+
+	// Schema holds the schema for the template version.
+	//
+	// +optional
+	SchemaRef *LocalObjectReference `json:"schemaRef,omitempty"`
+
+	// OriginalUISchema holds the original UI schema for the template version.
+	//
+	// +optional
+	OriginalUISchemaRef *LocalObjectReference `json:"originalUISchemaRef,omitempty"`
+
+	// UISchema holds the UI schema for the template version.
+	//
+	// +optional
+	UISchemaRef *LocalObjectReference `json:"uiSchemaRef,omitempty"`
+
+	// Removed indicate the template version is removed.
+	//
+	// +optional
+	Removed bool `json:"removed,omitempty"`
```

Comment:
How does it work?

Reply:
While the imported version is deleted in the vcs, will set removed to true, since this version may already used in resource.